lib/AST/ASTMangler.cpp

```diff
@@ -3838,7 +3838,7 @@ CanType ASTMangler::getDeclTypeForMangling(
 
   auto &C = decl->getASTContext();
   if (decl->isInvalid()) {
-    if (isa<AbstractFunctionDecl>(decl)) {
+    if (isa<AbstractFunctionDecl>(decl) || isa<EnumElementDecl>(decl)) {
```

Comment:
Maybe subscripts as well?

Reply:
Looks like the demangler doesn't currently require a function type for a subscript, but yeah let's also handle that, good catch!